### PR TITLE
fix tracking

### PR DESCRIPTION
**Pull request by tessalt**
uses correct event keys and calls `track` on view of products and collections

@richgilbank

### CODE REVIEW

File: src/components/product.js [hunk 1]
```diff
@@ -375,6 +375,19 @@ export default class Product extends Component {
   }
 
   /**
+   * get info about product to be sent to tracker
+   * @return {Object}
+   */
+  get trackingInfo() {
+    return {
+      id: this.id,
+      name: this.model.selectedVariant.productTitle,
```
Comment (richgilbank) [inline src/components/product.js]: Should it be `name` or `title`? It's `title` in product-set.

File: src/components/product.js [hunk 2]
```diff
@@ -375,6 +375,19 @@ export default class Product extends Component {
...
+      sku: null,
```
Comment (richgilbank) [inline src/components/product.js]: Is SKU required?

File: src/components/product.js (hunk 1 above)
Comment (zqureshi) [inline src/components/product.js]: I think it's called `name` here
https://github.com/Shopify/widget-framework/blob/master/app/assets/javascripts/framework/lib/tracking.js.coffee#L151

File: src/components/product.js (hunk 2 above)
Comment (tessalt) [inline src/components/product.js]: idk in original buy button it was null if undefined

Comment (zqureshi) [inline src/components/product.js]: is it not available anymore?

reference: https://github.com/Shopify/widget-framework/blob/master/app/assets/javascripts/framework/lib/tracking.js.coffee#L150

File: src/utils/track.js [hunk 3]
```diff
@@ -39,6 +39,15 @@ export default class Tracker {
     }
   }
 
+  trackComponent(type, properties) {
+    switch(type) {
+      case 'product':
+        return this.track('Viewed Product', properties);
+      case 'collection':
+        return this.track('Viewed Collection', propreties);
```
Comment (zqureshi) [inline src/utils/track.js]: It's `Viewed Collections` here
https://github.com/Shopify/widget-framework/blob/master/app/assets/javascripts/framework/lib/tracking.js.coffee#L118

**Comment by richgilbank** [comment]
:+1:

**Pull request merged**